### DISCUSSION TITLE
Empty tours in solution

### DISCUSSION THREAD

**Discussion by AngryAntoine** [reinterpretcat/vrp#160] [Q&A] [answered]
Hello,

I recently started using version 1.24.0, but I often encounter solutions with no tours. For example, when I run the solver consecutively on the same problem, some solutions include tours, while others do not. In these cases, stops are returned as unassigned. I used the public image to run the solver. In contrast, this issue did not occur in the older version (1.19.0), where I never observed missing tours in solutions.

I would be grateful for any information regarding this behavior.

**Comment by reinterpretcat** [comment]
Hi, do you have example file to reproduce the issue?

**Comment by AngryAntoine** [comment]
Hi,

This is what I used to get the solution (only changing the solution file name between the solver calls)
`vrp-cli solve pragmatic problem.json -o first_run_solution.json --max-time=5 --search-mode=broad --heuristic=default --max-generations=10 --init-size=1 --log`

The problem
[problem.json](https://github.com/user-attachments/files/17658149/problem.json)

And this is what I've got
[first_run_solution.json](https://github.com/user-attachments/files/17658147/first_run_solution.json)
[second_run_solution.json](https://github.com/user-attachments/files/17658148/second_run_solution.json)

**Reply by reinterpretcat** [reply, marked as answer]
The reason why you're getting such strange results is your objective:

```json
  "objectives": [
    {
      "type": "minimize-cost"
    },
    {
      "type": "minimize-unassigned"
    }
  ]
```
Please note, that the order is lexicographical, which means that the order of objectives defines their priority.  Essentially, here you're saying that `minimize-cost` is the most important objective. During the search, sometimes the solver is capable to find solution when all jobs are unassigned and cost is zero (e.g. in 10 generations). Theoretically, this is an optimal solution according to the objective..

In most of the cases, your top objective is minimize amount of jobs unassigned while cost is kept low. So, just swap objectives and it should work. If you would like to have amount of vehicles minimized (before the cost), then simply remove `objectives` as it is default behavior.

**Reply by AngryAntoine** [reply]
Thank you very much! It worked perfectly.
